### PR TITLE
Fix duplicate after 2 clip moves, 2 undos, and 1 redo

### PR DESCRIPTION
**Pull request by traprog**
The current redo implementation stores information in producers. This makes difficult to deal with consecutive moves with the same selection.

I made a bunch of `QList` member variables, so that information stays within each undo command object.

This is highly reliant on the assumption that the order the producers are inserted into `newSelection` is the same as all the integers are inserted into the `QList`s.

It's working well on my tests and clears all the reports posted in the forum's v22.01 beta test thread that lead to reverting the initial attempt.

### CODE REVIEW

File: src/commands/timelinecommands.cpp
Comment (bmatherly) [inline src/commands/timelinecommands.cpp:500]: The following constants seem to be unused after this change:
* kNewTrackIndexProperty
* kClipIndexProperty
* kShotcutInProperty
* kShotcutOutProperty

So we could remove them from shotcut_mlt_properties.h

File: src/commands/timelinecommands.cpp
Comment (bmatherly) [inline src/commands/timelinecommands.cpp:502]: Question for @ddennedy:

I see the following properties being used in other parts of the code:
* kPlaylistStartProperty
* kTrackIndexProperty

I do not fully understand the purpose of those producer properties. Should they continue to be set here since they seem to be accessed in other parts of the code?

File: src/commands/timelinecommands.cpp
Comment (bmatherly) [inline src/commands/timelinecommands.cpp:533]: Maybe kPlaylistStartProperty and kTrackIndexProperty need to be set on the clip here. I am unsure.

**Review by bmatherly — CHANGES_REQUESTED** [review]
Testing went good for me. I have some questions about the properties.

File: src/commands/timelinecommands.cpp
Comment (traprog) [reply src/commands/timelinecommands.cpp:500]: Yeah, They can be removed. 
I thought that it'd be better to do it in another commit because strictly speaking these deletions contribute nothing to fixing the current bug.

File: src/commands/timelinecommands.cpp
Comment (ddennedy) [reply src/commands/timelinecommands.cpp:502]: Yes these must be updated when a clip moves.

File: src/commands/timelinecommands.cpp
Comment (traprog) [reply src/commands/timelinecommands.cpp:533]: We can set them by passing values from `QList`s to producer properties.
But they won't be updated during undos because `MoveClipCommand::undo()` uses `UndoHelper`, and I don't see any lines in `UndoHelper` that contain `kPlaylistStartProperty` and `kTrackIndexProperty`.

File: src/commands/timelinecommands.cpp
Comment (traprog) [reply src/commands/timelinecommands.cpp:502]: I restored lines that set values to these properties.
I think here is fine instead of later.

Comment (traprog) [reply src/commands/timelinecommands.cpp:502]: I just checked about `kTrackIndexProperty`.

When a user moves a clip from V2 (track index 0) to V1 (track index 1), kTrackIndexProperty remains 0 because it only registers a value retrieved from `m_model.findClipByUuid`, not a value calculated by `trackIndex + m_trackDelta`.
`kTrackIndexProperty` is relatively new entry in `shotcut_mlt_properties.h`, so the only affected place is `glaxnimateproducerwidget.cpp`

I think this is a separate issue. I haven't even looked into the issue of `kPlaylistStartProperty` and `UndoHelper`. Maybe this one is fine since it's been around like 5 years. We can merge this pull request after additional critiques if there is any.

File: src/commands/timelinecommands.cpp
Comment (bmatherly) [reply src/commands/timelinecommands.cpp:500]: It is fine as a separate commit, but I would like that commit to  be part of this PR so that we don't forget.

Comment (traprog) [reply src/commands/timelinecommands.cpp:500]: OK. I don't mind deletions being part of this pull request.
It's going to be better to squash all the commits, though. I don't want "Restore kTrackIndexProperty kPlaylistStartProperty" a part of the commit history.

File: src/commands/timelinecommands.cpp
Comment (bmatherly) [reply src/commands/timelinecommands.cpp:502]: I do not have any further comments, but I would like to see a thumbs up from Dan before we resolve this comment.

**Review by ddennedy — CHANGES_REQUESTED** [review]
I do not have any specific change at this time. However, I think this change is risky and requires much more analysis and testing before merging. So, I want to block merging until I am comfortable with it.

**Review by ddennedy — APPROVED** [review]

**Pull request merged**
